### PR TITLE
[Gutenberg] - Remove Gallery V1 and experimental flags

### PR DESCRIPTION
**Pull request by geriux**
**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6973
- https://github.com/WordPress/gutenberg/pull/63285

To test:

- Test adding a Gallery block to a post

## Regression Notes
1. Potential unintended areas of impact
It affects only the editor.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and existing automated tests for the Gallery block.

3. What automated tests I added (or what prevented me from doing so)
No tests were added since this is only removing the old Gallery block format and a few experimental flags.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/292">25.2</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/Models/BlockEditorSettingElement+CoreDataProperties.swift [hunk 1]
```diff
@@ -11,12 +11,6 @@ enum BlockEditorSettingElementTypes: String {
     }
 }
 
-enum BlockEditorExperimentalFeatureKeys: String {
-    case galleryWithImageBlocks
-    case quoteBlockV2
-    case listBlockV2
```
Comment (dcalhoun) [inline WordPress/Classes/Models/BlockEditorSettingElement+CoreDataProperties.swift:17]: Were the quote and list values already unused by the editor?

**Review by dcalhoun — APPROVED** [review]
Test plan succeeded for me. I noted oddities I observed in https://github.com/wordpress-mobile/gutenberg-mobile/pull/6973#pullrequestreview-2167369822.

File: WordPress/Classes/Models/BlockEditorSettingElement+CoreDataProperties.swift (hunk 1 above)
Comment (geriux) [reply WordPress/Classes/Models/BlockEditorSettingElement+CoreDataProperties.swift:17]: Correct! I found one remaining [reference](https://github.com/WordPress/gutenberg/blob/73719c4281e49548d3ba9456a8a0064967cd4875/packages/react-native-editor/ios/GutenbergDemo/GutenbergViewController.swift#L373-L374) that I'll remove and link to this PR so we can include it as well.

**Pull request merged**